pkg/csi_driver/node.go

```diff
@@ -69,22 +69,15 @@ func (s *nodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublish
 	}
 
 	// Validate volume attributes
-	attr := req.GetVolumeAttributes()
+	attr := req.GetVolumeContext()
 	if err := validateVolumeAttributes(attr); err != nil {
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
 	// Check if mount already exists
 	mounted, err := s.isDirMounted(targetPath)
-	if err != nil {
-		// On Windows the mount target must not exist
-		if goOs == "windows" {
-			if !os.IsNotExist(err) {
-				return nil, err
-			}
-		} else {
-			return nil, err
-		}
+	if err != nil && !os.IsNotExist(err) {
```

Comment:
This changes ensures that CSI sanity test for node publish works if the target path does not exist.
https://github.com/kubernetes-csi/csi-test/blob/master/pkg/sanity/node.go#L341
The test attempts to node publish on a target path "/csi/mount/target", and the original driver code fails the node publish if the target path is not found. Is this behavior correct?

Reply:
The CSI spec says that the driver is responsible for creating the target path. However, Kubernetes does it wrong and creates the publish path, and we were mistakenly relying on this behavior: https://github.com/kubernetes/kubernetes/issues/75535

So adding this check is good and will also be good when we actually change the behavior in Kubernetes.

Reply:
I see, so does this would mean, if the target path does not exist the driver should explicitly create the dir and then proceed with mount.
Can this be taken up as a separate issue once this base patch is in?

Reply:
That's correct. It's fine to do this as a followup since Kubernetes is incorrectly creating it for us anyway. But we plan to remove that behavior in 1.20.

Reply:
I have put a TODO to explicitly create the target path if not present.